Open Judge System/Web/OJS.Web/Areas/Contests/Controllers/CompeteController.cs

```diff
@@ -100,8 +100,8 @@ public void ValidateContest(Contest contest, bool official)
         {
             if (contest == null ||
                 contest.IsDeleted ||
-                !contest.IsVisible ||
-                !this.IsUserAdminOrLecturerInContest(contest))
+                !(contest.IsVisible ||
```

Comment:
Can you insert the negation (!) in the braces, the code is more readable when you don't have to negate a complex condition. It should be:
contest == null ||
contest.IsDeleted ||
(!contest.IsVisible &&
    !this.IsUserAdminOrLecturerInContest(contest)))